server/src/main/java/org/opensearch/node/Node.java

```diff
@@ -1267,6 +1269,12 @@ public Node start() throws NodeValidationException {
         nodeService.getSearchBackpressureService().start();
         nodeService.getTaskCancellationMonitoringService().start();
 
+        if (FeatureFlags.isEnabled(REMOTE_STORE)) {
+            IndicesService indicesService = injector.getInstance(IndicesService.class);
+            RemoteStorePressureService pressureService = injector.getInstance(RemoteStorePressureService.class);
+            indicesService.setPressureService(pressureService);
```

Comment:
This can create inconsistencies with the contract. Constructor of `IndicesService` does not accept RemoteStorePressureService, that means if I were to create instance of IndicesService, I always have to remember to call `setPressureService()` afterwards. 

2 ways I can think of to tackle this:
1. Explicitly create instance of `RemoteStorePressureService` in Node.java (that means remove @Inject annotation from its constructor) and pass it to constructor of `IndicesService` - I prefer this
2. Change constructor of `IndicesService` to accept `RemoteStorePressureService` but it can be nullable. So, caller can be aware that one of the fields is an instance of pressure service and can use setter method to actually set the pressureService instance. While using it in the methods of IndicesService, we need to check if pressureService is null.

Reply:
There will be only 1 instance of `IndicesService` for a Node. We would not face the issue of setting the pressureService instance again. However, I am debating in my head for keeping this or having a tracker factory that can be created during node bootstrap and passed over to the required classes.

Reply:
But the contract is still broken. We are expecting one mandatory field to be provided as setter.

Reply:
@sachinpkale, @ashking94 I was hoping that we would be able to pull the `RemoteStoreStatsTrackerFactory` instance in `IndicesService` via Guice. However, `IndicesService` is instantiated before `RemoteStoreStatsTrackerFactory` is available to be pulled.

Hence, I have implemented something similar to the approach# 1 proposed by @sachinpkale above (i.e. not use Guice to instantiate `RemoteStoreStatsTrackerFactory` and instead pass the explicitly-created `RemoteStoreStatsTrackerFactory` instance to the constructor of `IndicesService`). Please review this and let me know if you have any recommendations / improvements on this front.

Thanks!